Prepare release 2.7.10 (#4844)
open-formulieren/open-forms: CHANGELOG.rst

Changelog
=========

2.7.10 (2024-11-22)
===================

Periodic bugfix release

* [:backend:`4732`] Fixed CSP issues for Expoints/other analytics.
* [:backend:`4745`] Fixed missing registration variable to the Objects API with all
  the attachment URLs.
* [:backend:`4810`] Fixed uppercase component variable values turing lowercase.

>   There are manual actions required.

Reviewer: Please include the steps to execute too, see for example https://open-forms.readthedocs.io/en/stable/archive/changes-v2.html#id68

> * [:backend:`4810`] Fixed uppercase component variable values turing lowercase.
>   There are manual actions required.

Reviewer: ```suggestion
* [:backend:`4810`] Fixed uppercase selectboxes options being lowercased if the component is
  in a step that's being skipped. See the above instructions on how to patch existing forms.
```

> * [:backend:`4823`] Fixed uploaded files with leading or trailing whitespaces in the
>   filename.

Reviewer: ```suggestion
* [:backend:`4823`] Fixed uploading files with leading or trailing whitespace in the
  filename.
```

> * [:backend:`4727`] Fixed crash when a user defined variable was changed to an array
>   datatype.

Reviewer: ```suggestion
* [:backend:`4727`] Fixed a crash in the form designer UI when a user defined variable was 
  changed to an array datatype.
```

Changelog
=========

2.7.10 (2024-11-22)
===================

Upgrade procedure
-----------------

* ⚠️ Ensure that the form variables initial values for selectboxes components match with
  their respective component default values.
  

>   After upgrading to 2.7.10, run the ``app/bin/fix_selectboxes_component_default_values.py``

Reviewer: ```suggestion
  After upgrading to 2.7.10, run the ``/app/bin/fix_selectboxes_component_default_values.py``
```
I think the working directory by default is `/app`, so using absolute paths is most robust.

  script. This script automatically fixes all initial values of the existing selectboxes
  component form variables.

  .. code-block:: bash

       # in the container via ``docker exec`` or ``kubectl exec``:
       python src/manage.py app/bin/fix_selectboxes_component_default_values.py

> Periodic bugfix release

Reviewer: This one typically goes right after the version number/date heading, so above the upgrade procedure :)
